homeassistant/components/esphome/sensor.py

```diff
@@ -104,7 +105,7 @@ def native_value(self) -> datetime | str | None:
             return None
         if self.device_class is SensorDeviceClass.TIMESTAMP:
             return dt_util.utc_from_timestamp(state_float)
-        return f"{state_float:.{self._static_info.accuracy_decimals}f}"
+        return float(state_float)
```

Comment:
```suggestion
        return state_float
```

`state_float` is already typed to `float` so you shouldn't need to wrap it in `float` again